src/multipage.jl

```diff
@@ -0,0 +1,186 @@
+ansi_moveup(n::Int) = string("\e[", n, "A")
+const ansi_movecol1 = "\e[1G"
+
+"""
+    play(framestack::AbstractVector{T}; kwargs...) where {T<:AbstractArray}
+    play(arr::T, dim::Int; kwargs...)
+
+Play a video of a framestack of image arrays, or 3D array along dimension `dim`.
+
+Control keys:
+- `p` or `space-bar`: pause/resume
+- `f`, `←`(left arrow), or `↑`(up arrow): step backward
+- `b`, `→`(right arrow), or `↓`(down arrow): step forward
+- `ctrl-c` or `q`: exit
+
+kwargs:
+
+- `fps::Real=30`
+
+# Examples
+
+```julia
+julia> using TestImages, ImageShow
+
+julia> img3d = testimage("mri-stack") |> collect;
+
+julia> ImageShow.play(img3d, 3)
+
+julia> framestack = [img3d[:, :, i] for i in axes(img3d, 3)];
+
+julia> ImageShow.play(framestack)
+```
+"""
+function play(framestack::AbstractVector{<:AbstractArray}; fps::Real=30, paused=false)
+    nframes = length(framestack)
+
+    # vars
+    frame_idx = 1
+    actual_fps = 0
+    should_exit = false
+
+    function render_frame(frame_idx, actual_fps; first_frame)
+        frame = framestack[frame_idx]
+        cols, rows = size(frame)
+
+        if !first_frame
+            print(ansi_moveup(2), ansi_movecol1)
+        end
+        println("Frame: $frame_idx/$nframes FPS: $(round(actual_fps, digits=1))", " "^5)
+        println("exit: ctrl-c. play/pause: space-bar. seek: arrow keys")
+
+        display(frame)
+    end
+    render_frame(frame_idx, actual_fps; first_frame=true)
+
+    keytask = @async begin
+        while !should_exit
+            control_value = read_key()
+
+            if control_value == :CONTROL_BACKWARD
+                frame_idx = max(frame_idx-1, 1)
+            elseif control_value == :CONTROL_FORWARD
+                frame_idx = min(frame_idx+1, nframes)
+            elseif control_value == :CONTROL_PAUSE
+                paused = !paused
+            elseif control_value == :CONTROL_EXIT
+                should_exit = true
+            elseif control_value == :CONTROL_VOID
+                nothing
+            else
+                error("Control value $control_value not recognized.")
+            end
+        end
+    end
+
+    try
+        last_frame_idx = frame_idx
+
+        while !should_exit && 1<= frame_idx <= nframes
+            # when paused, only update the frame when last_frame_idx changes, i.e., only when
+            # user hit arrow keys.
```

Comment:
In [the original PR](https://github.com/JuliaImages/ImageInTerminal.jl/pull/43), the same constant image is repeatedly rendered even if in "paused" status. This PR changed that behavior so that the plot panel count number doesn't increase infinitely.